### PR TITLE
Adding support for proper CRLF

### PR DESCRIPTION
**Pull request by sharadrsoni**
Window's Notepad doesn't recognize `\n` linebreak it has to be `\r\n`

### CODE REVIEW

**Comment by jim-parry** [comment]
This is going to mess things up for all non-notepad usage, isn't it?
The better strategy could be to use notepad++ instead of notepad.

**Comment by gxgpet** [comment]
Or to use the existing `PHP_EOL`.

**Comment by sharadrsoni** [comment]
> This is going to mess things up for all non-notepad usage, isn't it?
> The better strategy could be to use notepad++ instead of notepad.

I have checked on notepad++ and other editors, it is working fine.

**Comment by sharadrsoni** [comment]
> Or to use the existing `PHP_EOL`. But yeah, Notepad++ should do the trick, too.

PHP_EOL is defined as "\n"

**Comment by gxgpet** [comment]
> PHP_EOL is defined as "\n"

Well, if you are running two different types of OS on your server and your own machine, then nothing can help, but a more advanced text editor/viewer.

**Comment by sharadrsoni** [comment]
> > PHP_EOL is defined as "\n"
> 
> Well, if you are running two different types of OS on your server and your own machine, then nothing can help, but a more advanced text editor/viewer.

No, I am using windows only, using WAMP stack.
My thought is we have a default application for notepad by Windows OS, we need to support them, we can't just tell to use any other notepad editor.
This is a log file, ideally shared across multiple OS and can be viewed by layman too, for them they don't know what is notepadd++.

I hope you are understanding my point.

**Comment by gxgpet** [comment]
Just made a test on my local env:

```
C:\xampp\php>php.exe -r "echo strlen(PHP_EOL);"
2
```

Are you sure about that `PHP_EOL` being `\n`?

**Comment by sharadrsoni** [comment]
> PHP_EOL

Correct because there are 2 characters only. i.e. `\n`

**Comment by narfbg** [comment]
> Correct because there are 2 characters only. i.e. `\n`

`\n` is a single character.

Also: https://devblogs.microsoft.com/commandline/extended-eol-in-notepad/

I would just close this, but `PHP_EOL` seems like a decent suggestion.

**Comment by sharadrsoni** [comment]
> > Correct because there are 2 characters only. i.e. `\n`
> 
> `\n` is a single character.
> 
> Also: https://devblogs.microsoft.com/commandline/extended-eol-in-notepad/
> 
> I would just close this, but `PHP_EOL` seems like a decent suggestion.

My Bad, reason: On ctrl + click on `PHP_EOL` PhpStrom shows `Core_d.php` where it only shows `\n` but while evaluating an expression runtime, I got the value of '\r\n'

I have made necessary changes, but still, for my project, I am going to use `/r/n` as it is harmless for Windows and Linux [to my understanding].

My use-case is, clients, download the logs and open in notepad in Windows, for them it looks like a mess, hence have to use `/r/n` as it is not harming in Linux too.

Thanks for sharing the Link of notepad, but that feature is not yet enabled, and to so one need to change the registry which is not a cup of tea for every layman.

**Comment by gxgpet** [comment]
That is because PhpStorm's internal documentation relies on some hardcoded stubs (which are available on the Github as well for editing), and not on the configured PHP interpreter.

**Comment by narfbg** [comment]
> Thanks for sharing the Link of notepad, but that feature is not yet enabled, and to so one need to change the registry which is not a cup of tea for every layman.

The thing has worked with just `\n` for the entirety of CodeIgniter's existence. We're not just now going to change it because a few people don't have a third-party software's patch yet ... :)

File: system/core/Log.php [hunk 1]
```diff
@@ -249,11 +249,11 @@ public function write_log($level, $msg)
 	 * @param	string	$level 	The error level
 	 * @param	string	$date 	Formatted date string
 	 * @param	string	$message 	The log message
-	 * @return	string	Formatted log line with a new line character '\n' at the end
+	 * @return	string	Formatted log line with a new line character '\r\n' at the end
 	 */
 	protected function _format_line($level, $date, $message)
 	{
-		return $level.' - '.$date.' --> '.$message."\n";
+		return $level.' - '.$date.' --> '.$message. PHP_EOL;
```
Comment (narfbg) [inline system/core/Log.php]: No spaces around the concatenation operator please, same applies for the other file as well.

File: system/core/Log.php [hunk 2]
```diff
@@ -249,11 +249,11 @@ public function write_log($level, $msg)
 	 * @param	string	$level 	The error level
 	 * @param	string	$date 	Formatted date string
 	 * @param	string	$message 	The log message
-	 * @return	string	Formatted log line with a new line character '\n' at the end
+	 * @return	string	Formatted log line with a new line character '\r\n' at the end
```
Comment (narfbg) [inline system/core/Log.php]: ... and now this isn't necessarily true.

I guess it's better just to remove it, so it only says newline character.

**Pull request merged**